kitsune/search/v2/base.py

```diff
@@ -306,27 +310,46 @@ class SumoSearch(SumoSearchInterface):
     class inherits, relevant to the documents the child class is searching over.
     """
 
-    results_per_page: int = settings.SEARCH_RESULTS_PER_PAGE
-    total: int = 0
-    hits: list[AttrDict] = dfield(default_factory=list)
-    results: list[dict] = dfield(default_factory=list)
+    total: int = dfield(default=0, init=False)
+    hits: list[AttrDict] = dfield(default_factory=list, init=False)
+    results: list[dict] = dfield(default_factory=list, init=False)
+    last_key: Union[int, slice, None] = dfield(default=None, init=False)
+
+    query: str = ""
+    default_operator: str = "AND"
+
+    def __len__(self):
+        return self.total
+
+    @overload
+    def __getitem__(self, key: int) -> dict:
+        ...
+
+    @overload
+    def __getitem__(self, key: slice) -> list[dict]:
+        ...
+
+    def __getitem__(self, key):
+        if self.last_key is None or self.last_key != key:
+            self.run(key=key)
+        if isinstance(key, int):
+            return self.results[0]
```

Comment:
Should this be `0`?

Reply:
`self.run(key=key)` has already done the offset in the ES query, so `self.results` will be a list with only one item (if the key was an int), agreed this is a little confusing, will add a comment clarifying